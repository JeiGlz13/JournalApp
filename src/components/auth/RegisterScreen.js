import React from 'react';
import {Link} from 'react-router-dom'; 
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { removeError, setError } from '../../actions/ui';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';


export const RegisterScreen = () => {
    const dispatch = useDispatch();
    const {msgError} = useSelector(state => state.ui);

    const [formValues, handleInputChange] = useForm({
            name: '',
            email: '',
            password: '',
            password2: ''
        });

        const {name, email, password, password2} = formValues;

        const handleRegister = (e)=>{
            e.preventDefault();
            if(isFormValid()){
                dispatch(startRegisterWithEmailPasswordName(email, password, name));
            }
        }

        const isFormValid = ()=>{
            if(name.trim().length === 0){
                dispatch(setError('Name is required'));
                return false;
            }else if(!validator.isEmail(email)){
                dispatch(setError('Email is not valid'));
                return false;
            }else if(password !== password2 || password.length < 6){
                dispatch(setError('Password should be at least 6 characters'));
                return false;
            }
            dispatch(removeError());
            return true;
        }

    return (
        <div>
            <h3 className = "auth__title">Register</h3>

            <form onSubmit = {handleRegister} className = "animate__animated animate__fadeIn animate-faster auth__form">
                {   msgError &&
                    <div className= "auth__alert-error">
                        {msgError}
                    </div>
                }

                <input className = "auth__input" type = "text" placeholder = "Name" name = "name" value = {name} onChange = {handleInputChange} autoComplete="off" />
                <input className = "auth__input" type = "text" placeholder = "Email" name = "email" value = {email} onChange = {handleInputChange} autoComplete="off" />
                <input className = "auth__input" type = "password" placeholder = "Password" name = "password" value = {password} onChange = {handleInputChange} />
                <input className = "auth__input" type = "password" placeholder = "Confirm Password" name = "password2" value = {password2} onChange = {handleInputChange} />

                <button className = "btn btn-primary btn-block" type = "submit">
                    Login
                </button>
                
                <div className = "auth__social-networks">

                    <Link className = "link" to = "/auth/login"> 
                        Already registered?
                    </Link>
                </div>
            </form>
        </div>
    )
}
