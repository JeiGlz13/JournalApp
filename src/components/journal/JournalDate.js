import React from 'react'

export const JournalDate = ({date}) => {
  return (
    <div className="date-container">
        <div className="date-month">
            {
                date.format('MMMM')
            }
        </div>

        <span className="date-day">
            {
                date.format('D')
            }
        </span>
        
        <div className="pb-2 px-2 border-l border-r border-b rounded-b flex justify-between date-year">
            <span className="text-xs font-bold">
                {
                    date.format('ddd')
                }
            </span>
            <span className="text-xs font-bold">
                {
                    date.format('YYYY')
                }
            </span>
        </div>
    </div>
  )
}
