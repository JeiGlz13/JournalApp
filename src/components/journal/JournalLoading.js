import React from 'react';
import Typical from 'react-typical';
import relojArena from '../../assets/relojArena.gif';

export const JournalLoading = () => {
  return (
    <div className='loading__screen'>
        <img src= {relojArena} alt='loading' />
        <Typical
            steps={['Loading...', 1200, '', 550]}
            loop={Infinity}
            wrapper="h1"
        />
    </div>
  )
}
