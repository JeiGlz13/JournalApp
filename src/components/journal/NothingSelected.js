import React from 'react'

export const NothingSelected = () => {
    return (
        <div className = "nothing__mainContent">
            <p>
                Select something
                <br/>
                or crete an Entry
            </p>

            <i className = "far fa-star fa-4x marginTop-2" ></i>
        </div>
    )
}
