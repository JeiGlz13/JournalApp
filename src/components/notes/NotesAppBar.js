import moment from 'moment';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';
import { JournalDate } from '../journal/JournalDate';

export const NotesAppBar = () => {
    const dispatch = useDispatch();
    const {active: note} = useSelector(state => state.notes)
    const handleSave = ()=>{
        dispatch(startSaveNote(note));
    }

    const noteDate = moment(note.date);

    const handlePictureUpload = ()=>{
        document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e)=>{
        const file = (e.target.files[0]);
        if(file){
            dispatch(startUploading(file));
        }
    }

    
    return (
        <div className = "notes__appBar">
            <JournalDate date={noteDate} />

            <input type = "file" accept='image/*' id = 'fileSelector' 
            style = {{display: 'none'}} onChange = {handleFileChange} />
            
            <div className='btns-container'>
                <button className = " btn-header"
                onClick = {handlePictureUpload}>
                    <i className="fa-solid fa-image"></i>
                    Add picture
                </button>

                <button className = " btn-header"
                onClick = {handleSave}>
                    <i className="fa-solid fa-floppy-disk"></i>
                    Save
                </button>
            </div>
        </div>
    )
}
