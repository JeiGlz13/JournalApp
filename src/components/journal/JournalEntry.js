import React from 'react';
import moment from 'moment';
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({id, date, title, body, url}) => {

    const noteDate = moment(date);
    const dispatch = useDispatch();

    const handleEntryClick = ()=>{
        dispatch(activeNote(id, {
            date, title, body, url
        }));
    }
    return (
        <div className = "journal__entry pointer animate__animated animate__fadeIn animate-faster" 
            onClick = {handleEntryClick}>
            {
                url &&
                <div className="journal__entryPicture"
                    style = {{
                        backgroundSize: 'cover',
                        backgroundImage: `url(${url})`
                    }}>
                </div>
            }

            <div className = "journal__entryBody">
                <p className = "journal__entryTitle">
                    {title}
                </p>
                <p className = "journal__entryContent">
                    {body}
                </p>
            </div>

            <div className = "journal__entryDateBox">
                <span>{noteDate.format('ddd')}</span>
                <h4>{noteDate.format('Do')}</h4>
            </div>
        </div>
    )
}
