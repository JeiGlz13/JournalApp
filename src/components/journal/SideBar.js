import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../../actions/auth';
import { startNewNote } from '../../actions/notes';
import {JournalEntries} from './JournalEntries';

export const SideBar = () => {

    const dispatch = useDispatch();
    const {name} = useSelector(state => state.auth)

    const handleLogout = ()=>{
        dispatch(startLogout());
    }

    const handleAddNote = ()=>{
        dispatch(startNewNote());
    }
    return (
        <aside className = "journal__sideBar">
            <div className = "journal__sideBar-navBar">
                <h3 className = "marginTop-2">
                    <i className = "far fa-moon"></i>
                    <span> {name}</span>
                </h3>

                <button className = "btn"
                onClick = {handleLogout}>
                    Logout
                </button>
            </div>

            <div className = "journal__newEntry"
                onClick = {handleAddNote}>
                <i className = "far fa-calendar-plus fa-5x" ></i>
                <p className = "marginTop-2">New Entry</p>
            </div>

            <JournalEntries/>
        </aside>
    )
}
