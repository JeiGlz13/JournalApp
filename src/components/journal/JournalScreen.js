import React from 'react';
import { useSelector } from 'react-redux';
import {SideBar} from '../journal/SideBar';
import {NothingSelected} from './NothingSelected';
import {NoteScreen} from '../notes/NoteScreen';

export const JournalScreen = () => {
    const {active} = useSelector(state => state.notes)
    return (
        <div className = "journal__main-content animate__animated animate__fadeIn animate-faster">
            <SideBar />
            <main>
                {
                    (active)
                    ? (<NoteScreen/>)
                    : ( <NothingSelected/>)
                }
                {/* */}
                
            </main>
        </div>
    )
}
