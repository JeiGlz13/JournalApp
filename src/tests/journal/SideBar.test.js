import React from 'react';
import {mount} from 'enzyme';
import { Provider } from 'react-redux';
import {MemoryRouter} from 'react-router-dom';

import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import { login, startLogout } from '../../actions/auth';
import { AppRouter } from '../../routers/AppRouter';
import { act } from '@testing-library/react';
import Swal from 'sweetalert2';
import { startNewNote } from '../../actions/notes';
import { SideBar } from '../../components/journal/SideBar';


jest.mock('../../actions/auth', ()=>({
    startLogout: jest.fn()
}));

jest.mock('../../actions/notes', ()=>({
    startNewNote: jest.fn()
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '1',
        name: 'Jeisosn'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes:{
        active: null,
        notes: []
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store = {store}>
            <SideBar/>
    </Provider>
    );

describe('Pruebas en <SideBar/>', () => {
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe llamar el startLogout', () => {
        wrapper.find('button').simulate('click');
        expect(startLogout).toHaveBeenCalled();
    });
    
    test('Debe llamar el startNewNote', () => {
        wrapper.find('.journal__newEntry').simulate('click');
        expect(startNewNote).toHaveBeenCalled();
    })
    
})
