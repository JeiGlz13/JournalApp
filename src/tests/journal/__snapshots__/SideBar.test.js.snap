// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SideBar/> Debe mostrarse correctamente 1`] = `
<aside
  className="journal__sideBar"
>
  <div
    className="journal__sideBar-navBar"
  >
    <h3
      className="marginTop-2"
    >
      <i
        className="far fa-moon"
      />
      <span>
         
        Jeisosn
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal__newEntry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="marginTop-2"
    >
      New Entry
    </p>
  </div>
  <div
    className="journal__entries"
  />
</aside>
`;
